fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -279,7 +316,8 @@ public void writeData(long id, int block, @Nonnull byte[] value) {
     @Override
     public void deleteFile(@Nonnull String name) throws IOException {
```

Comment:
This isn't part of this PR, but it looks like every call in `FDBDIrectory` is synchronous -- meaning you `join()` or `asyncToSync()` before returning. Hopefully the methods are never called by any asynchronous code?